nft_ingester/src/api/api_impl.rs

```diff
@@ -382,6 +386,35 @@ where
         Ok(json!(res))
     }
 
+    pub async fn get_nft_editions(&self, payload: GetNftEditions) -> Result<Value, DasApiError> {
+        let label = "get_nft_editions";
+        self.metrics.inc_requests(label);
+        let latency_timer = Instant::now();
+
+        let pagination = Pagination {
+            limit: payload.limit.unwrap_or(DEFAULT_LIMIT as u32).into(),
+            page: payload.page,
+            before: payload.before,
+            after: payload.after,
+            cursor: payload.cursor,
+        };
+        Self::validate_basic_pagination(&pagination, self.max_page_limit)?;
+        let mint_address = Pubkey::from_str(&payload.mint)
```

Comment:
can't this parameter be deserialized by `serde` as I wrote above?

Reply:
We use convert/validate functions. `validate_pubkey(key)?;`
I have replaced to use it.

Reply:
You need to convert the parameter each time you want a `Pubkey` because the plain `string` seems to doesn't make much sense. My suggestion is to use a deserializer by `serde` and pass already freshly-baked, nice and already `Pubkey` and catch errors early instead of propagating them through the code because we are used to using `String` instead of `Pubkey`. The code is yours, though, so the choice is as well.

Reply:
Yes, I agree with your point. My latest impl validate `Publkey`  almost immediately and sends an throw an error if value is not valid. 
`let mint_address = validate_pubkey(payload.mint)?;`

Reply:
https://linear.app/mplx/issue/MTG-1441/implement-a-custom-serde-serializerdeserializer-to-automatically